[AMD] use DC method for linalg.eigh

Mellonta opened the pull request:
Summary: Jacobi method has larger numerical errors, see D64997718, use divide-and-conquer method instead.

Test Plan: CI

Differential Revision: D65786796

Mellonta commented:
@pytorchbot label "topic: not user facing"

jianyuh reviewed (approved):
BTW, what's the differences between syevj vs. syevd? Jacobi method vs. QR algorithm?

Mellonta commented:
> BTW, what's the differences between syevj vs. syevd? Jacobi method vs. QR algorithm?

j stands for Jacobi and d stands for divide-and-conquer.

Mellonta committed:
[AMD] use DC method for linalg.eigh (#140327)

Summary:

Jacobi method has larger numerical errors, see D64997718, use divide-and-conquer method instead.

Test Plan: CI

Reviewed By: jianyuh

Differential Revision: D65786796
